### PR TITLE
Adds a resin turret

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
See CL

## Why It's Good For The Game

Xemno content

## Changelog
:cl:
add: New xeno sticky resin turret, can be selected by right clicking the action
imageadd: Sprites by Ciel/kyranova
/:cl:

### CODE REVIEW

File: code/modules/xenomorph/xeno_structures.dm [hunk 1]
```diff
@@ -1044,3 +1046,12 @@ TUNNEL
...
 	newshot.fire_at(hostile, src, null, ammo.max_range, ammo.shell_speed)
+
+/obj/structure/xeno/resin/xeno_turret/sticky
+	name = "resin sticky resin turret"
+	icon = 'icons/Xeno/acidturret.dmi'
+	icon_state = XENO_TURRET_STICKY_ICONSTATE
+	desc = "A menacing looking construct of resin, it seems to be alive. It fires resin against intruders."
+	light_initial_color = LIGHT_COLOR_PURPLE
+	ammo = /datum/ammo/xeno/sticky
+	firerate = 6
```
Comment (BraveMole) [inline code/modules/xenomorph/xeno_structures.dm:1057]: is this working with the turret autofire? it though only multiple of 20 worked

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by AndroidMK2** [comment]
Would be good  if it was much cheaper than the acid turret.

**Review by Kuro020 — APPROVED** [review]

File: code/modules/xenomorph/xeno_structures.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/xenomorph/xeno_structures.dm:1057]: idk will change to 5

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 2]
```diff
@@ -1155,15 +1155,34 @@
...
+/datum/action/xeno_action/activable/build_turret/update_button_icon()
+	button.overlays.Cut()
+	button.overlays += image(initial(turret_type.icon), initial(turret_type.icon_state))
+	return ..()
+
+/datum/action/xeno_action/activable/build_turret/alternate_action_activate()
+	INVOKE_ASYNC(src, .proc/pick_turret)
+	return COMSIG_KB_ACTIVATED
+
+/datum/action/xeno_action/activable/build_turret/proc/pick_turret()
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:1179]: Autodocced, but the default placing doesnt have a keybind, its rarely used and i plan on removing this anyway in an upcoming change so is it ok if i just leave it kbless

**Review by BraveMole — APPROVED** [review]

**Comment by Tristan63** [comment]
https://youtu.be/eVTXPUF4Oz4

**Pull request merged**